File: VB/T191208/MyBarManager.vb
Imports DevExpress.XtraBars
Imports DevExpress.XtraBars.Utils
Imports System.ComponentModel
Imports System.Reflection

Namespace T191208
    Public Class MyBarManager
        Inherits BarManager

        Public Sub New()
            MyBase.New()
        End Sub
        Public Sub New(ByVal components As IContainer)
            MyBase.New(components)
        End Sub
        '
        'This method is hidden from IntelliSense
        'since 'Helper' class is designed for internal use
        'Avoid using it in your app
        '
        <EditorBrowsable(EditorBrowsableState.Never)> _
        Public Function GetHelper() As BarManagerHelpers
            Return Me.Helper
        End Function
        Public Sub SetCustomizeSelectedLink(ByVal link As BarItemLink)
            Me.SelectionInfo.CustomizeSelectedLink = link
        End Sub

        'We do not recommend using Reflection 
        Public Sub SetSelectedItem(ByVal item As BarItem)
            Dim selItemField As FieldInfo = Me.SelectionInfo.GetType().GetField("selectedItem", System.Reflection.BindingFlags.NonPublic Or System.Reflection.BindingFlags.Instance)
            selItemField.SetValue(Me.SelectionInfo, item)
        End Sub
    End Class
End Namespace

File: CS/T191208/MyBarManager.cs
using DevExpress.XtraBars;
using DevExpress.XtraBars.Utils;
using System.ComponentModel;
using System.Reflection;

namespace T191208 {
    public class MyBarManager : BarManager {
        public MyBarManager()
            : base() {
        }
        public MyBarManager(IContainer components)
            : base(components) {
        }
        //
        //This method is hidden from IntelliSense
        //since 'Helper' class is designed for internal use
        //Avoid using it in your app
        //
        [EditorBrowsable(EditorBrowsableState.Never)]
        public BarManagerHelpers GetHelper() {
            return this.Helper;
        }
        public void SetCustomizeSelectedLink(BarItemLink link) {
            this.SelectionInfo.CustomizeSelectedLink = link;
        }

        //We do not recommend using Reflection 
        public void SetSelectedItem(BarItem item) {
            FieldInfo selItemField = this.SelectionInfo.GetType().GetField("selectedItem", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance);
            selItemField.SetValue(this.SelectionInfo, item);
        }
    }
}
